src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -339,13 +361,34 @@ def deleteBlocks(self):
         logging.info("Checking if there are block rules to be deleted...")
 
         # Get list of blocks that can be deleted
-        blockDict = self.findDeletableBlocks.execute(transaction=False)
+        # blockDict = self.findDeletableBlocks.execute(transaction=False)
+        completedBlocksList = self.getCompletedBlocks.execute(transaction=False)
+
+        if not completedBlocksList:
+            logging.info("No candidate blocks found for rule deletion.")
+            return
 
-        if not blockDict:
-            logging.info("No candidate blocks found for rule deletion")
+        logging.info("Found %d completed blocks", len(completedBlocksList))
+        logging.debug("Full completedBlocksList: %s", pformat(completedBlocksList))
+
+        deletableWfsDict = self.getDeletableWorkflows.execute()
+
+        if not deletableWfsDict:
+            logging.info("No workflows in terminal state found (Including child workflows). Skipping block level rule deletion in the current run.")
```

Comment:
I think this "including child workflows" might be more confusing than clarifying.

Reply:
fixed